tests/python/TestBoundingBox.py

```diff
@@ -0,0 +1,479 @@
+
+        # Check success
+        self.assertEqual(response[0]["AddBoundingBox"]["status"], 0)
+
+    def addBoundingBoxwithImage(self, db, numBoxes, imgprops=None):
+        all_queries = []
+        imgs_arr = []
+
+        fd = open("../test_images/brain.png", 'rb')
+        imgs_arr.append(fd.read())
+        fd.close()
+
+        img = {}
+        img["properties"] = imgprops
+        img["format"] = "png"
+        img["_ref"] = 12
+
+        query = {}
+        query["AddImage"] = img
+
+        all_queries.append(query)
+
+        basename = imgprops["name"] + "_bb_"
+        for x in range(0, numBoxes):
+            bb_coords = {}
+            bb_coords["x"] = x*10
+            bb_coords["y"] = x*10
+            bb_coords["height"] = 100
+            bb_coords["width"] = 100
+
+            bbprops = {}
+            bbprops["name"] = basename + str(x)
+            bb = {}
+            bb["properties"] = bbprops
+            bb["rectangle"] = bb_coords
+            bb["image"] = 12
+
+            bb_query = {}
+            bb_query["AddBoundingBox"] = bb
+            all_queries.append(bb_query)
+
+        response, res_arr = db.query(all_queries, [imgs_arr])
+
+        self.assertEqual(len(response), numBoxes + 1)
+        self.assertEqual(response[0]["AddImage"]["status"], 0)
+        for i in range(0, numBoxes):
+            self.assertEqual(response[i+1]["AddBoundingBox"]["status"], 0)
+
+    def test_addBoundingBox(self):
+        db = vdms.vdms()
+        db.connect(hostname, port)
+
+        all_queries = []
+
+        for i in range(0, self.number_of_inserts):
+            bb_coords = {}
+            bb_coords["x"] = i
+            bb_coords["y"] = i
+            bb_coords["height"] = 512
+            bb_coords["width"] = 512
+
+            props = {}
+            props["name"] = "my_bb_" + str(i)
+
+            bb = {}
+            bb["properties"] = props
+            bb["rectangle"] = bb_coords
+
+            query = {}
+            query["AddBoundingBox"] = bb
+
+            all_queries.append(query)
+
+        response, img_array = db.query(all_queries)
+
+        self.assertEqual(len(response), self.number_of_inserts)
+        for i in range(0, self.number_of_inserts):
+            self.assertEqual(response[i]["AddBoundingBox"]["status"], 0)
+
+    def test_findBoundingBox(self):
+        db = vdms.vdms()
+        db.connect(hostname, port)
+
+        prefix_name = "find_my_bb_"
+
+        for i in range(0, self.number_of_inserts):
+            props = {}
+            props["name"] = prefix_name + str(i)
+            self.insertBoundingBox(db, props=props)
+
+        all_queries = []
+
+        for i in range(0, self.number_of_inserts):
+            constraints = {}
+            constraints["name"] = ["==", prefix_name + str(i)]
+
+            results = {}
+            results["list"] = ["name"]
+
+            bb_params = {}
+            bb_params["constraints"] = constraints
```

Comment:
I think it would end up being more complicated than it needs to be right now? If we're running into a timing issue, we can discuss that. But I think we'd need to change how we store the coordinates if we wanted to use a range query (right now we store starting coords and width/height ... we'd want to probably store start x/end x and start y/end y if we wanted to work with ranges)